tests/functional/serving/test_trials_resource.py

```diff
@@ -0,0 +1,287 @@
+        """
+        Tests that if an unknown parameter is specified in
+        the query string, an error is returned even if the experiment doesn't exist.
+        """
+        expected_error_message = 'Parameter "unknown" is not supported. ' \
+                                 'Expected one of [\'ancestors\', \'status\', \'version\'].'
+
+        response = client.simulate_get('/trials/a?unknown=true')
+
+        assert response.status == "400 Bad Request"
+        assert response.json == {'title': 'Invalid parameter',
+                                 'description': expected_error_message}
+
+        add_experiment(name='a', version=1, _id=1)
+
+        response = client.simulate_get('/trials/a?unknown=true')
+
+        assert response.status == "400 Bad Request"
+        assert response.json == {'title': 'Invalid parameter',
+                                 'description': expected_error_message}
+
+    def test_trials_for_latest_version(self, client):
+        """Tests that it returns the trials of the latest version of the experiment"""
+        add_experiment(name='a', version=1, _id=1)
+        add_experiment(name='a', version=2, _id=2)
+
+        add_trial(experiment=1, id_override="00")
+        add_trial(experiment=2, id_override="01")
+        add_trial(experiment=1, id_override="02")
+        add_trial(experiment=2, id_override="03")
+
+        response = client.simulate_get('/trials/a')
+
+        assert response.status == "200 OK"
+        assert response.json == [{'id': '01'}, {'id': '03'}]
+
+    def test_trials_for_specific_version(self, client):
+        """Tests specific version of experiment"""
+        add_experiment(name='a', version=1, _id=1)
+        add_experiment(name='a', version=2, _id=2)
+        add_experiment(name='a', version=3, _id=3)
+
+        add_trial(experiment=1, id_override="00")
+        add_trial(experiment=2, id_override="01")
+        add_trial(experiment=3, id_override="02")
+
+        # Happy case
+        response = client.simulate_get('/trials/a?version=2')
+
+        assert response.status == "200 OK"
+        assert response.json == [{'id': '01'}]
+
+        # Version doesn't exist
+        response = client.simulate_get('/trials/a?version=4')
+
+        assert response.status == "404 Not Found"
+        assert response.json == {
+            'title': 'Experiment Not Found',
+            'description': 'Experiment "a" has no version "4"'
+        }
+
+    def test_trials_for_all_versions(self, client):
+        """Tests that trials from all ancestors are shown"""
+        add_experiment(name='a', version=1, _id=1)
+        add_experiment(name='a', version=2, _id=2)
+        add_experiment(name='a', version=3, _id=3)
+
+        add_trial(experiment=1, id_override="00")
+        add_trial(experiment=2, id_override="01")
+        add_trial(experiment=3, id_override="02")
+
+        # Happy case default
+        response = client.simulate_get('/trials/a?ancestors=true')
```

Comment:
It may be worth testing `trials/a?ancestors=true,version=2` to make sure the trials are fetched from the current experiment as the root.

Reply:
Just saw the other test. Forget it. :)